src/app/domain/backup/queries/index.ts

```diff
@@ -2,23 +2,26 @@ import { Q, QueryDefinition } from 'cozy-client'
 
 const DOCTYPE_FILES = 'io.cozy.files'
 
-export const buildFilesQuery = (dirId: string): QueryDefinition => {
+export const buildFilesQuery = (deviceId: string): QueryDefinition => {
   return Q(DOCTYPE_FILES)
     .where({
-      dir_id: dirId,
-      type: 'file'
+      type: 'file',
+      'metadata.backupDeviceIds': {
+        $elemMatch: { $eq: deviceId }
+      }
     })
-    .indexFields(['dir_id', 'type'])
-    .select(['name', 'dir_id', 'type'])
+    .indexFields(['type', 'metadata.backupDeviceIds'])
```

Comment:
You should invert the order. Always the most restrictive before. I think this is documented somewhere here https://docs.cozy.io/en/tutorials/data/advanced/#indexes-performances-and-design